postgresql/connection.go

```diff
@@ -0,0 +1,95 @@
+package postgresql
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"os"
+
+	_ "github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/dialers/postgres" // blank import
+	"github.com/joho/godotenv"
+
+	"github.com/benchttp/server/benchttp"
+)
+
+type computedStatsService struct {
+	db *sql.DB
+}
+
+func NewComputedStatsService(idleConn, maxConn int) (benchttp.ComputedStatsService, error) {
+	connectionInfo, err := getConnectionInfoFromEnvVariables()
+	if err != nil {
+		return nil, err
+	}
+
+	dbURI := fmt.Sprintf("host=%s user=%s password=%s dbname=%s sslmode=disable",
+		connectionInfo.host,
+		connectionInfo.user,
+		connectionInfo.password,
+		connectionInfo.dbName)
+
+	db, err := sql.Open("cloudsqlpostgres", dbURI)
+	if err != nil {
+		return nil, ErrDatabaseConnection
+	}
+
+	err = db.Ping()
+	if err != nil {
+		return nil, ErrDatabasePing
+	}
+
+	db.SetMaxIdleConns(idleConn)
+	db.SetMaxOpenConns(maxConn)
+
+	return &computedStatsService{db}, nil
+}
+
+func (b *computedStatsService) Close() {
+	b.db.Close()
+}
+
+type connectionInfo struct {
+	host     string
+	user     string
+	password string
+	dbName   string
+}
+
+func getConnectionInfoFromEnvVariables() (connectionInfo, error) {
```

Comment:
[suggestion] I'd recommend to handle this in `main`, and inject the retrieved values into `benchmarkService` from there: it'll make it more independent and testable!